tokio/src/io/driver/mod.rs

```diff
@@ -374,6 +355,36 @@ impl Inner {
 
         Ok(())
     }
+
+    /// shutdown the dispatcher.
+    fn shutdown(&self) -> bool {
+        let mut io = self.io_dispatch.write().unwrap();
```

Comment:
forgive me if this is a dumb question:

is it ok for this `unwrap` to cause panics?  Isn't it better to use a `Result`/`io::Error` ?

Reply:
It is fine, see [RwLock::write()](https://doc.rust-lang.org/stable/std/sync/struct.RwLock.html#method.write)